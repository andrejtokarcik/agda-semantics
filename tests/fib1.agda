module fib1 where

data ℕ : Set where
  Z : ℕ 
  S : ℕ -> ℕ 


_+_ : ℕ -> ℕ ->  ℕ
n + Z   = n
n + S m = S (n + m)

one : ℕ 
one = S Z

fib : ℕ -> ℕ
fib Z         = one
fib (S Z)     = one
fib (S (S n)) = fib n + fib (S n)
