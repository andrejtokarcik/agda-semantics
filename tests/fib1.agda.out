<k>
    .K
</k>
<mgu>
    subst(.KList)
</mgu>
<ctx-stack>
    .List
</ctx-stack>
<sig>
    Fun ( "_+_" ) |-> ( #symVariable(4) : Data ( "8469" ) ) -> ( #symVariable(
      5) : Data ( "8469" ) ) -> Data ( "8469" )
    Fun ( "fib" ) |-> ( #symVariable(55) : Data ( "8469" ) ) -> Data ( "8469" 
      )
    Fun ( "one" ) |-> Data ( "8469" )
    Data ( "8469" ) |-> Set ( 0 )
    Con ( "S" ) |-> ( #symVariable(3) : Data ( "8469" ) ) -> Data ( "8469" )
    Con ( "Z" ) |-> Data ( "8469" )
</sig>
<ctx>
    .Map
</ctx>
<meta>
    .Map
</meta>
<ncat>
    Name ( "8469" ) |-> Data ( "8469" )
    Name ( "S" ) |-> Con ( "S" )
    Name ( "Z" ) |-> Con ( "Z" )
    Name ( "_+_" ) |-> Fun ( "_+_" )
    Name ( "fib" ) |-> Fun ( "fib" )
    Name ( "one" ) |-> Fun ( "one" )
</ncat>
<fun>
    Fun ( "_+_" ) |-> 'fc(((#symFunClauses(17) Var ( "n" )) Con ( "Z" )) = Var
       ( "n" ),, ((#symFunClauses(46) Var ( "n" )) (Con ( "S" ) Var ( "m" ))) 
      = (Con ( "S" ) ((Fun ( "_+_" ) Var ( "n" )) Var ( "m" ))))
    Fun ( "fib" ) |-> 'fc((#symFunClauses(61) Con ( "Z" )) = Fun ( "one" ),, (
      #symFunClauses(71) (Con ( "S" ) Con ( "Z" ))) = Fun ( "one" ),, (
      #symFunClauses(106) (Con ( "S" ) (Con ( "S" ) Var ( "n" )))) = ((Fun ( 
      "_+_" ) (Fun ( "fib" ) Var ( "n" ))) (Fun ( "fib" ) (Con ( "S" ) Var ( 
      "n" )))))
    Fun ( "one" ) |-> (Con ( "S" ) Con ( "Z" ))
</fun>
